src/js/index.js

```diff
@@ -0,0 +1,157 @@
+
+const $menuList = $("#menu-list");
+
+const $nav = $("nav");
+const $categoryTitle = $("#category-title");
+
+const store = {
+  setLocalStorage(value) {
+    localStorage.setItem("menu", JSON.stringify(value));
+  },
+
+  getLocalStorage() {
+    return JSON.parse(localStorage.getItem("menu"));
+  },
+};
+
+function App() {
+  this.menu = {
+    espresso: [],
+    frappuccino: [],
+    blended: [],
+    teavana: [],
+    desert: [],
+  };
+  this.currentCategory = "espresso";
+  this.soldout = false;
+
+  this.init = () => {
+    if (store.getLocalStorage()) {
+      this.menu = store.getLocalStorage();
+    }
+    render();
+  };
+
+  const render = () => {
+    const template = this.menu[this.currentCategory]
+      .map((item, index) => {
+        return `
+        <li data-menu-id="${index}" class="menu-list-item d-flex items-center py-2">
+          <span class="w-100 pl-2 menu-name">
+            ${item.name}
+          </span>
+          <button
+            type="button"
+            class="bg-gray-50 text-gray-500 text-sm mr-1 menu-sold-out-button"
+          >
+            품절
+          </button>
+          <button
+            type="button"
+            class="bg-gray-50 text-gray-500 text-sm mr-1 menu-edit-button"
+          >
+            수정
+          </button>
+          <button
+            type="button"
+            class="bg-gray-50 text-gray-500 text-sm menu-remove-button"
+          >
+            삭제
+          </button>
+        </li>
+      `;
+      })
+      .join("");
+
+    $menuList.innerHTML = template;
+    countMenu();
+  };
+
+  const addMenu = () => {
+    if (!$menuNameInput.value) {
+      alert("값을 입력하세요.");
+      return;
+    }
+
+    this.menu[this.currentCategory].push({ name: $menuNameInput.value });
+    store.setLocalStorage(this.menu);
+    render();
+    $menuNameInput.value = "";
+  };
+
+  const editMenu = (e) => {
+    const menuId = e.target.closest("li").dataset.menuId;
+    const result = window.prompt("메뉴명을 수정하세요.");
+    this.menu[this.currentCategory][menuId].name = result;
+    store.setLocalStorage(this.menu);
+    e.target.parentNode.firstElementChild.innerText = result;
+  };
+
+  const deleteMenu = (e) => {
+    if (confirm("정말 삭제하시겠습니까?")) {
+      const menuId = e.target.parentNode.dataset.menuId;
+      this.menu[this.currentCategory].splice(menuId, 1);
+      store.setLocalStorage(this.menu);
+      $menuList.removeChild(e.target.parentNode);
+
+      countMenu();
+    }
+  };
+
+  const countMenu = () => {
+    const count = $menuList.childElementCount;
+    $menuCount.innerHTML = `총 ${count}개`;
+  };
+
+  const soldOutMenu = (e) => {
+    const menuId = e.target.closest("li").dataset.menuId;
+    this.soldout = !this.soldout;
+
+    if (this.soldout) {
+      e.target.parentNode.firstElementChild.classList.add("sold-out");
+    }
+
+    console.log(this.soldout);
+  };
+
+  $menuForm.addEventListener("submit", (e) => {
+    e.preventDefault();
+    addMenu();
+  });
```

Comment:
이벤트 핸들러 등록을 할때 이름을 붙여서 하는게 좋을 것 같다. 익명함수가 아니라 이름을 붙이고 넘기면 좋을것 같다
```suggestion
  $menuForm.addEventListener("submit", function addMenuClick(e) {
    e.preventDefault();
    addMenu();
  });
```
또는
```suggestion
const addMenuClick = (e) => {
    e.preventDefault();
    addMenu();
  }
  
// 함수 자체를 넘기므로 뒤에 괄호가 있으면 안됨
$menuForm.addEventListener("submit", addMenuClick);
```
이벤트리스너 안에서 this를 쓰는건 위험한 행위이니 주의하도록 한다